src/tests/ftest/cart/threaded_server.c

```diff
@@ -107,29 +107,39 @@ int main(int argc, char **argv)
 	int			status = 0;
 	int			rc;
 	int			i;
+	int                     thread_count;
 
 	rc = d_log_init();
 	assert(rc == 0);
 
 	rc = crt_init("manyserver", CRT_FLAG_BIT_SERVER | CRT_FLAG_BIT_AUTO_SWIM_DISABLE);
 	if (rc != 0) {
 		printf("Could not start server, rc = %d", rc);
-		return -1;
+		goto log_fini;
 	}
 
 	rc = crt_proto_register(&my_proto_fmt_threaded_server);
 	if (rc != 0) {
 		printf("Could not register rpc protocol , rc = %d", rc);
-		return -1;
+		goto fini;
 	}
 
 	rc = crt_context_create(&crt_ctx);
-	assert(rc == 0);
-	for (rc = 0; rc < NUM_THREADS; rc++)
-		pthread_create(&thread[rc], NULL, progress, &status);
+	if (rc != 0) {
+		printf("Failed to create context: " DF_RC "\n", DP_RC(rc));
+		goto fini;
+	}
+
+	for (thread_count = 0; thread_count < NUM_THREADS; thread_count++) {
+		rc = pthread_create(&thread[thread_count], NULL, progress, &status);
+		if (rc != 0) {
+			printf("Failed to create thread: %d, exiting\n", rc);
```

Comment:
Do we want to format the error code?
```suggestion
			printf("Failed to create thread:  %d (%s), exiting\n", rc, strerror(rc));
```

Reply:
Fixed